jujugui/static/gui/src/app/models/models.js

```diff
@@ -714,6 +718,37 @@ YUI.add('juju-models', function(Y) {
     },
 
     /**
+      Generate a name for this service with an incremented number if needed.
+
+      @method _generateServiceName
+      @param {String} charmName The charm name.
+      @param {String} charmId the full charm id.
+      @returns {String} The name for the service.
+    */
+    _generateServiceName: function(charmName, charmId) {
+      var highestId = 0;
+      this.each(function(service) {
```

Comment:
To avoid listings in the future on laaaarge bundles, maybe a map of map[serviceName, maxCount] could be used instead. It would also solve race conditions when 2 or more are working on the same bundle.

But that's for the future. Maybe add a TODO here.